tidy3d/web/api/container.py

```diff
@@ -649,32 +639,57 @@ def delete(self) -> None:
         for _, job in self.jobs.items():
             job.delete()
 
-    def real_cost(self) -> float:
-        """Get the sum of billed costs for each task associated with this batch."""
+    def real_cost(self, verbose: bool = True) -> float:
+        """Get the sum of billed costs for each task associated with this batch.
+
+        Parameters
+        ----------
+        verbose : bool = True
+            Whether to log the cost and helpful messages.
+
+        Returns
+        -------
+        float
+            Billed cost for the entire :class:`.Batch`.
+        """
         real_cost_sum = 0.0
         for _, job in self.jobs.items():
-            cost_job = job.real_cost()
+            cost_job = job.real_cost(verbose=False)
             if cost_job is not None:
                 real_cost_sum += cost_job
-        return real_cost_sum or None
 
-    def estimate_cost(self) -> float:
+        real_cost_sum = real_cost_sum or None  # convert to None if 0
+
+        if real_cost_sum and verbose:
+            console = get_logging_console()
+            console.log(f"Total billed flex credit cost: {real_cost_sum:1.3f}.")
+        return real_cost_sum
+
+    def estimate_cost(self, verbose: bool = True) -> float:
         """Compute the maximum FlexCredit charge for a given :class:`.Batch`.
 
-        Returns
-        -------
-        float
-            Estimated maximum cost for each Union[:class:`.Simulation`, :class:`.HeatSimulation`]
-            associated with given :class:`.Batch`.
+        Parameters
+        ----------
+        verbose : bool = True
+            Whether to log the cost and helpful messages.
 
         Note
         ----
         Cost is calculated assuming the simulation runs for
-        the full ``run_time``. If early shut-off is triggered, the cost is adjusted proporionately.
+        the full ``run_time``. If early shut-off is triggered, the cost is adjusted proportionately.
 
         Returns
         -------
         float
             Estimated total cost of the tasks in FlexCredits.
         """
-        return sum(job.estimate_cost() for _, job in self.jobs.items())
+        batch_cost = sum(job.estimate_cost(verbose=False) for _, job in self.jobs.items())
+
+        if verbose:
+            console = get_logging_console()
+            if batch_cost is not None and batch_cost > 0:
+                console.log(f"Maximum FlexCredit cost: {batch_cost:1.3f} for the whole batch.")
+            else:
+                console.log("Could not get estimated batch cost!")
```

Comment:
maybe say why (eg it might not have finished computing, wait and try again?)

Reply:
Actually when it comes to estimated cost, it should always work now. This is probably legacy code. This statement should not be reachable since if something goes wrong, the estimate cost should at least error. So yeah I don't really know of a case when this can be reached and so don't know how to improve the message - but may just leave it just in case?